### DISCUSSION TITLE
SunbirdEd-mobile-app: Steps to create debug environment for Google sign-in

### DISCUSSION THREAD

**Discussion by chandu4002** [Sunbird-Ed/Community#647] [Bugs]
Currently Google sign-in webClinetId is working well on prod (playstore app), But in debug app with same webClinetId we are getting error.
Please provide the steps to create debug environment for Google sign-in process.

FYI, we are using **release-5.1.0** branch for Diksha app.

**Comment by swayangjit** [comment]
@chandu4002 You can create a new credential in google console project , give your default SHA-1 certificate fingerprint and same package name. It should work after that.

**Comment by chandu4002** [comment]
@swayangjit Thanks for the response.

As per your suggestion, we followed this process you mentioned. Now google sign-in is successfully completed but we are getting issue in the setSession method. Please see the below image for your reference line no: 237.

<img width="1079" alt="Screenshot 2023-11-06 at 6 57 04 PM" src="https://github.com/Sunbird-Ed/Community/assets/129141695/6b40308e-ec7b-4412-b953-667870e026a3">


We have checked the api call also, we are getting the following error 

**Resp: {
  "msg": "unable to create session"
}**

Note: we have created new project in Firebase with same package name and debug SHA-1 certificate fingerprint using the new webClinetId which we have created.

**Reply by swayangjit** [reply]
@chandu4002 please do clear the storage and check if its working or not

**Reply by chandu4002** [reply]
@swayangjit can you please explain which storage i need to clear ?

please mention do i need to change webClinetId from server side also as i am using new webClinetId ?

**Reply by swayangjit** [reply]
@chandu4002  freshly install the app or clear the device storage and you don't have to change the webClientId.

**Reply by chandu4002** [reply]
@swayangjit i have tried clearing the storage and freshly installing the app, in both ways it is not working ..
Behaving same like before..

Resp: **400 
{
"msg": "unable to create session"
}**

**Reply by swayangjit** [reply]
@chandu4002  Please let me know which method/piece of code/API return this error

**Reply by chandu4002** [reply]
@swayangjit please find the following code and API details for your reference..

**Code:**

<img width="1079" alt="Screenshot 2023-11-06 at 6 57 04 PM" src="https://github.com/Sunbird-Ed/Community/assets/129141695/71a76ad4-47c3-4e40-b13b-3bb21f86939b">

Line No: 237 
**await this.loginNavigationHandlerService.setSession(nativeSessionGoogleProvider, this.skipNavigation, InteractSubtype.GOOGLE)**

**Inside setSession():**

<img width="771" alt="Screenshot 2023-11-08 at 9 38 43 AM" src="https://github.com/Sunbird-Ed/Community/assets/129141695/9ce89683-44b3-4e8b-b28c-20aafa3e2bad">

Line No: 52
**await this.authService.setSession(webViewSession).toPromise();**


API details: 
**/google/auth/android**

Req:
**{
  "emailId": "xxxxxxxx@xxxx.com",
  "platform": null
}**

Resp:
**400
{
  "msg": "unable to create session"
}**

**Reply by swayangjit** [reply]
@chandu4002  platform should be android or ios can you check why its sending null

**Reply by chandu4002** [reply]
@swayangjit i have checked and this request with platform as 'null' is setting from sunbird-sdk library.
 **"@project-sunbird/sunbird-sdk": "5.0.0"**
 
 FYI,
 Following your suggestion, I am using new **webClineId** with new **google console account** with same **package name** and **SHA-1 fingerprint** which we fetched using the command (keytool -list -v \-alias androiddebugkey -keystore ~/.android/debug.keystore ) ..
 and i am using that new webClientId in the app code inside **signInWithGoogle()** method

**Reply by chandu4002** [reply]
@swayangjit Any update please?

**Reply by chandu4002** [reply]
@swayangjit Please update.

**Reply by swayangjit** [reply]
@chandu4002  API details:
/google/auth/android

Req:
{
"emailId": "[xxxxxxxx@xxxx.com](mailto:xxxxxxxx@xxxx.com)",
"platform": null
}

Resp:
400

This api is throwing error  can you check with portal team what this is returning 400

**Reply by chandu4002** [reply]
We are working on this, checking with portal team on the same.

**Comment by anisha147** [comment]
Hi @chandu4002 

Hope your queries were answered. If yes, could you please close this thread?

cc: @swayangjit @vinukumar-vs [@mohitga](https://github.com/mohitga)

**Reply by chandu4002** [reply]
Yes thanks, closing the thread..
